enterprise/api/v1/views.py

```diff
@@ -326,15 +327,33 @@ def contains_content_items(self, request, pk, course_run_ids, program_uuids):
 
         contains_content_items = True
         if course_run_ids:
-            contains_content_items = enterprise_customer_catalog.contains_content_items('key', course_run_ids)
+            contains_content_items = enterprise_customer_catalog.contains_courses(course_run_ids)
         if program_uuids:
-            contains_content_items = contains_content_items and enterprise_customer_catalog.contains_content_items(
-                'uuid',
-                program_uuids
+            contains_content_items = (
+                contains_content_items and
+                enterprise_customer_catalog.contains_programs(program_uuids)
             )
 
         return Response({'contains_content_items': contains_content_items})
 
+    @detail_route(url_path='courses/{}'.format(COURSE_KEY_URL_PATTERN))
+    def course_detail(self, request, pk, course_key):  # pylint: disable=invalid-name,unused-argument
+        """
+        Return the metadata for the specified course.
+
+        The course needs to be included in the specified EnterpriseCustomerCatalog
+        in order for metadata to be returned from this endpoint.
+        """
+        enterprise_customer_catalog = self.get_object()
+        course = enterprise_customer_catalog.get_course(course_key)
+        if not course:
+            raise Http404
+
+        context = self.get_serializer_context()
```

Comment:
any reason not to override get_serializer_context for this view set to always include enterprise_customer_catalog instead of adding it for each detail_route?

Reply:
Good idea.

Reply:
On second thought, that would require multiple calls to `self.get_object` resulting in multiple calls to the DB. I'm going to stick with this for now.